### ISSUE TITLE
`plotOptions` for specified series are not applied when using `setOptions`

### ISSUE THREAD

**Issue by goransle** [highcharts/highcharts#20716]
#### Expected behaviour
The series should be rendered the same way as when the option is provided in the `chart` function.

#### Actual behaviour
The chart is rendered using values from `plotOptions.series` and not `plotOptions.mapline`

#### Live demo with steps to reproduce
https://jsfiddle.net/goransle/k8ze4hq9/

#### Product version
HC 10+

#### Affected browser(s)
Chrome, Firefox, Safari

**Comment by hubertkozik** [comment]
The same issue is visible for e.g. `pie` series: https://jsfiddle.net/BlackLabel/trxje5fm/

**Comment by KacperMadej** [comment]
_Internal note:_
In short: `Highcharts.setOptions({ plotOptions: { series: ... }})` is not implemented/supported.

Looks like `Series`'s `setOptions` should have `typeOptions, plotOptions.series` order flipped in https://github.com/highcharts/highcharts/blob/a31cf6028b3fafb9b3aa35361c042e95c852814e/ts/Core/Series/Series.ts#L826-L827 but there is no difference between `defaultOptions.plotOptions.series` and `userOptions.plotOptions.series` (only `plotOptions.series` is used in the merge) so those 2 should be created instead of the existing one and merged in order `defaultOptions.plotOptions.series, defaultOptions.plotOptions[series.type], userOptions.plotOptions.series` (and the rest as is).

**Comment by ravindUwU** [comment]
I think I just ran into the same bug. Repro'd here: https://stackblitz.com/edit/ravinduwu-misc-2024-09-05-highcharts-setoptions-bug?file=index.html%3AL13

**Comment by hubertkozik** [comment]
@ravindUwU Yes, it looks to me, that your problem has the same root cause as mentioned by my colleague.

**Comment by ravindUwU** [comment]
Unsure if this is still an issue with the latest version of Highcharts; could the team verify this?

Also, could the stale bot _please_ be [configured to act on issues labelled unactionable](https://github.com/actions/stale#any-of-labels) (awaiting more info/non-reproducible/etc.) so that legitimate issues won't get closed simply because there wasn't any recent conversation around them 😒?

**Comment by KacperMadej** [comment]
Hi @ravindUwU 

This is still an issue.

Stale bot config could be improved, but as that is not related to the product, we'll make an internal task to discuss this.
